File: RMS UI/rms/src/app/services/table.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment.development';
import { table } from '../models/model';

@Injectable({
  providedIn: 'root'
})
export class TableService {

  constructor(private http: HttpClient) { }
  getAllTables(): Observable<any>
  {
    return this.http.get(`${environment.baseUrl}/api/Table/ListOfTables`);
  }
  
  showAvailableTables(): Observable<any>
  {
    return this.http.get(`${environment.baseUrl}/api/Table/AvailableTables`);
  }

  updateStatus(id: number, status: string):Observable<any>
  {
    return this.http.post<any>(`${environment.baseUrl}/api/Table/UpdateStatus/${id}/${status}`, null);
  }
  updateTable(id: number, table: table): Observable<any>
  {
    return this.http.post<any>(`${environment.baseUrl}/api/Table/UpdateTable?id=${id}`, table);
  }
  deleteTable(id: number): Observable<any>
  {
    return this.http.delete<any>(`${environment.baseUrl}/api/Table/DeleteTable/${id}`);
  }
}

File: RMS API/rms/Services/TableService.cs
using System;
using Repositories.TableRepo;
using Models.TableModel;
namespace Services.TableSvc
{
    public class TableService
    {
        public readonly ITableRepository _tableRepository;
        public TableService(ITableRepository tableRepository)
        {
            _tableRepository = tableRepository;
        }
        public Table GetTableById(int Id)
        {
            return _tableRepository.GetTableById(Id);
        }
        public List<Table> ListOfTables()
        {
            return _tableRepository.ListOfTables();
        }
        public List<Table> AvailableTables()
        {
            return _tableRepository.AvailableTables();
        }
        public Table UpdateTable(int Id, Table table)
        {
            return _tableRepository.UpdateTable(Id, table);
        }
        public Table DeleteTable(int Id)
        {
            return _tableRepository.DeleteTable(Id);
        }
        public bool UpdateStatus(int Id)
        {
            return _tableRepository.UpdateStatus(Id);
        }
        public bool UpdateStatus(int Id, string Status)
        {
            return _tableRepository.UpdateStatus(Id, Status);
        }
    }
}